### ISSUE TITLE
Able to enter a birth date into the future

### ISSUE THREAD

**Issue by elisekalstad** [civiform/civiform#1625]
**Describe the bug**
A resident completing an application could select a date far out into the future as their birthday

**To Reproduce**
1. Go to the form for the UDP test application
2. Complete the question sequence until you get to a birth date question
3. Select a date into the future
4. No error- able to select a date into the future

**Expected behavior**
I would expect the dates to be greyed out that are after today's date. Either that or an error message saying a birthdate needs to be in the past. This could be a new question type: "Birthdate" that prevents future birthdays. I would just want to be careful that the current date question type is good for residents completing other date-like questions that might actually go out into future dates.
![Birthdate into the future](https://user-images.githubusercontent.com/86082659/134586042-77f3e83d-f714-444b-99e9-4275db784479.png)

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by lindsayzad** [comment]
We should leave the general ability for date to be in the future, as that might be relevant for future program's questions.  I think the ask here is to allow the admin to optionally specify a possible min or max eligible date; i.e. only date's before this max date are allowed to be entered

**Comment by shanemc-goog** [comment]
For things like Birthday, it might make sense to have a "in the past" "in the future" condition that's a moving target with when NOW is.

It'd likely apply to other questions too.

**Comment by ettengererin** [comment]
Right, good point Shane. I think this is the idea for min and max date -- both would be able to set a specific date or a moving target for NOW.

Here are Mocks that Tanya made for this new and improved Date question
https://docs.google.com/presentation/d/1FgzHj47nyIKgy-Ars0FD6tbo0mSbvW8k71X1aqzprrY/edit?resourcekey=0-Qd3RHJN6ZuMNRE6ZQjneFg#slide=id.gf8553fdcd2_0_20

**Comment by elisekalstad** [comment]
This came up as an issue for the UDP program--applications are coming in with birthdates in 2021 (but they should be over 18 years old).

**Comment by elisekalstad** [comment]
Testing on Android phone--unable to hard type the date into the field. Would have to scroll month by month across long stretches of years.

**Comment by elisekalstad** [comment]
**Elise to review with project team on Tuesday**

**Comment by nb1701** [comment]
Assuming the default for current date questions should be "any past date" to "any future date"?  Or does "any past date" to "current date" make more sense?

**Comment by dkatzz** [comment]
> Assuming the default for current date questions should be "any past date" to "any future date"? Or does "any past date" to "current date" make more sense?

I think any past date to any future date should be the default

**Comment by dkatzz** [comment]
Related to https://github.com/civiform/civiform/issues/5677

**Comment by kpeterson-exygy** [comment]
Closing this ticket as a duplicate.
